rllib/offline/dataset_reader.py

```diff
@@ -228,14 +228,19 @@ def __init__(self, ds: ray.data.Dataset, ioctx: Optional[IOContext] = None):
                     if not self._ioctx.config.get("_disable_preprocessors", False)
                     else None
                 )
-            self._dataset.random_shuffle(seed=seed)
```

Comment:
`random_shuffle()` doesn't mutate datasets, so this statement doesn't do anything.

Reply:
Why does it not mutate the dataset?
The docstring says "Randomly shuffle the rows of this :class:`Dataset`".

After these changes, will the shuffling we do be deterministic without the seed?

Reply:
None of the `Dataset` APIs are in-place. When it says "Randomly shuffle the rows of this :class:Dataset", it means "Return a new dataset with the rows of this dataset shuffled."

Added a seed.